### PR TITLE
[DOCS] Reformat `flatten_graph` token filter

### PR DESCRIPTION
**Pull request by jrodewig**
Makes the following changes to the `flatten_graph` token filter docs:

* Rewrites description and adds Lucene link
* Adds detailed analyze example
* Adds analyzer example

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/flatten-graph-tokenfilter.asciidoc [hunk 1]
```diff
@@ -4,18 +4,232 @@
...
+
+[WARNING]
+====
+Flattening graphs containing multi-position tokens is a lossy process.
+Flattening collapses the side paths of a token graph into a linear path.
+
+If possible, avoid using graph token filters in
+<<analysis-index-search-time,index analyzers>>. Use the respective
+<<analysis-synonym-tokenfilter,`synonym`>> or
+<<analysis-word-delimiter-tokenfilter,`word_delimiter`>> filters instead.
```
Comment (romseygeek) [inline docs/reference/analysis/tokenfilters/flatten-graph-tokenfilter.asciidoc:24]: I think it would be good to reword this to strongly advise against using these at index time at all - generally speaking, filters that could produce graphs should be used at query time instead.  `flatten_graph` is a kind of shonky workaround for a problem that really should be fixed elsewhere.

**Review by romseygeek — CHANGES_REQUESTED** [review]
Thanks @jrodewig .  I think we need to more actively discourage people from using this filter, as it's a not-very-good solution for a problem that should be fixed by doing things at query time instead.

File: docs/reference/analysis/tokenfilters/flatten-graph-tokenfilter.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/flatten-graph-tokenfilter.asciidoc]: Thanks for the feedback @romseygeek!

With afcbebfb5ea49d3fe008d371fedd3ffb3e85cc0e, I reworded the warning to remove the recommendation for the `synonym` and `word_delimiter` filters. It also recommends that token graph filters only be used in search analyzers.

Let me know if you see any other needed changes!

**Review by romseygeek — APPROVED** [review]
LGTM, thanks @jrodewig

**Comment by jrodewig** [comment]
Thanks @romseygeek.

**Comment by jrodewig** [comment]
### Backport commits
master 2416036
7.x f0b9be8b1bb
7.7 13fbbbc3d7a
7.6 4998b807c6dda32b26e5d402f0741e5b20b20b5a
